### ISSUE TITLE
Inline eval no longer capable of using "it" to reference the output of the previous inline

### ISSUE THREAD

**Issue by gabrielbcn** [haskell/haskell-language-server#1577]
Obviously the inline code evaluation is amazing, but in previous versions I used to do this:

```
-- >>> example
-- >>> used it
-- >>> fresh it
-- \a. \x. (\y. a) x b
-- ["a","b","x","y"]
-- "c"
```

So I would use ```it``` to refer to the previous output, the same as in a ghci session,

Now the following happens:

```
-- >>> example
-- >>> used it
-- >>> fresh it
-- \a. \x. (\y. a) x b
-- Variable not in scope: it :: Term
-- Variable not in scope: it :: [Var]
```

it can still be sorted in many ways, a very ugly one with variables:

```
-- >>> example
-- >>> it1 = example
-- >>> used it1
-- >>> it2 = used it1
-- >>> fresh it2
-- \a. \x. (\y. a) x b
-- ["a","b","x","y"]
-- "c"
```

But it is really a pity because removes the versatility of ```it``` that just brings the previous output whatever it was.

Any chance to bring it back?

Thanks a lot

**Comment by jneira** [comment]
hi, thanks for the bug report, I ve translated the issue to the haskell-language-server repo as it is responsive for handling the eval feature
//cc @tittoassini

**Comment by tittoassini** [comment]
honestly, I wasn't aware that 'it' was ever supported :-)

**Comment by tittoassini** [comment]
I had a look, it is easy to add a:

it <- return e

after an expression 'e' is evaluated.


That's not quite what GHCi does. GHCi sets "it" to the result of "e" evaluation, not "e" itself.


So, for example: 

import System.CPUTime
*Ide.Arguments System.CPUTime> getCPUTime
2330167000000
*Ide.Arguments System.CPUTime> it
2330167000000
*Ide.Arguments System.CPUTime> it + it
4660334000000

However, we would set "it" to "getCPUTime" itself and then we would get:

-- >>> import System.CPUTime
-- >>> getCPUTime
-- >>> it
-- >>> it + it 
-- 25948208000000
-- 25953978000000
-- No instance for (Num (IO Integer)) arising from a use of ‘+’

Naturally, one could argue that is a more correct, or at least more consistent, behaviour.

If we wanted to do what GHCi does, we would need to check the type of the expression "e" and execute:

for a pure value:

it <- return e

for an IO a:

it <- a



What I don't get is how "it" might ever have been supported, maybe @pepeiborra can explain?

**Comment by jneira** [comment]
@tittoassini hi! thanks to take a look, want to note there is a open pull request trying to fix the issue: #1603
